redirect/client.js

```diff
@@ -0,0 +1,52 @@
+(function() {
+
+    function apply(context, template) {
+
+        /**
+         * Prevent redirect from occurring while in either the Template Editor or Campaign Editor.
+         */
+        if ((window.frameElement || {}).id === "siteEditorFrame"
+            || !context.targetPage.includes(window.location.hostname + window.location.pathname.replace(/\/$/gim, ""))) {
+            return;
+        }
+
+        return new Promise((resolve, reject) => {
+            if ((context.targetPage && context.urlForRedirect)
+                && window.location.href !== context.urlForRedirect) {
```

Comment:
The way it's written right now, the template will redirect from _any_ non-targetPage page. So for example, if targetPage is the _About Us_ page and urlForRedirect is the _Locations_ page on cumulus, navigating from a non-targetPage like homepage actually redirects to the _Locations_ page. We could ensure that we redirect only from targetPage by this: 
```suggestion
            if ((context.targetPage && context.urlForRedirect)
                && window.location.href === context.targetPage) {
```

Reply:
I'm not sure that's the case, since it shouldn't even reach this conditional if we're not on the target page, due to `!context.targetPage.includes(window.location.hostname + window.location.pathname.replace(/\/$/gim, ""))`

Reply:
See example:
<img width="1792" alt="Screen Shot 2020-11-09 at 11 21 10 AM" src="https://user-images.githubusercontent.com/32201252/98586532-ad60f100-227d-11eb-8019-eb616c2aed52.png">

Campaign runs on homepage, even with targetPage not set to it

Reply:
@nriserEG - This should be fixed with my latest commit.